spec/models/manageiq/providers/nuage/network_manager/vcr_specs/refresher_spec.rb

```diff
@@ -44,5 +64,124 @@ def assert_table_counts
     expect(ExtManagementSystem.count).to eq(1)
     expect(NetworkGroup.count).to eq(2)
     expect(SecurityGroup.count).to eq(1)
+    expect(CloudSubnet.count).to eq(2)
+    expect(FloatingIp.count).to eq(0)
+    expect(NetworkPort.count).to eq(0)
+    expect(NetworkRouter.count).to eq(0)
+  end
+
+  def assert_ems
+    expect(@ems.network_groups.count).to eq(2)
+    expect(@ems.security_groups.count).to eq(1)
+    expect(@ems.cloud_subnets.count).to eq(2)
+
+    expect(@ems.network_groups.map(&:ems_ref))
+      .to match_array([network_group_ref1, network_group_ref2])
+    expect(@ems.security_groups.map(&:ems_ref))
+      .to match_array([security_group_ref])
+    expect(@ems.cloud_subnets.map(&:ems_ref))
+      .to match_array([cloud_subnet_ref1, cloud_subnet_ref2])
+  end
+
+  def assert_network_groups
+    g1 = NetworkGroup.find_by(:ems_ref => network_group_ref1)
+    expect(g1).to have_attributes(
+      :name                   => "Ansible-Test",
+      :cidr                   => nil,
+      :status                 => "active",
+      :enabled                => nil,
+      :ems_id                 => @ems.id,
+      :orchestration_stack_id => nil,
+      :type                   => "ManageIQ::Providers::Nuage::NetworkManager::NetworkGroup"
+    )
+    expect(g1.cloud_subnets.count).to eq(0)
+    expect(g1.security_groups.count).to eq(0)
+
+    g2 = NetworkGroup.find_by(:ems_ref => network_group_ref2)
+    expect(g2).to have_attributes(
+      :name                   => "XLAB",
+      :cidr                   => nil,
+      :status                 => "active",
+      :enabled                => nil,
+      :ems_id                 => @ems.id,
+      :orchestration_stack_id => nil,
+      :type                   => "ManageIQ::Providers::Nuage::NetworkManager::NetworkGroup"
+    )
+    expect(g2.cloud_subnets.count).to eq(2)
```

Comment:
Here you are just testing the counts - please also check that the cloud subnets and security groups are actually the correct ones, i.e. `g1`, `s1` and `s2` below.

Reply:
Nice one, thanks.